test/supplemental_genbank_tests/genbank_upload_full_test.py

```diff
@@ -222,44 +221,109 @@ def _load_expected_data(self, json_path):
+        with open(json_path, "w") as outfile:
+            json.dump(dictionary, outfile)
+
+    def _get_blob_id(self, handle_id):
+        handles = self.hs.hids_to_handles([handle_id])
+        blob_id = handles[0]['id']
+        return blob_id
+
+    def _download_file_from_blobstore(self, handle_id):
+        output_dir = self.cfg['scratch'] + "/" + str(uuid.uuid4())
+        os.makedirs(output_dir)
+        file_ret = self.dfuClient.shock_to_file(
+            {
+                'handle_id': handle_id,
+                'file_path': output_dir,
+                'unpack': 'unpack',
+            }
+        )
+        return file_ret['file_path']
+
+    def _calculate_md5sum(self, file_path):
+        md5 = hashlib.md5()
+        with open(file_path, "rb") as f:
+            for chunk in iter(lambda: f.read(4096), b""):
+                md5.update(chunk)
+        return md5.hexdigest()
+
     def _retrieve_provenance(self, provenance):
-        # make a copy to avoid modifying the original provenance
+        # make a deep copy to avoid modifying the original provenance
         provs = [prov.copy() for prov in provenance]
         for prov in provs:
             for key in ["time", "epoch"]:
                 prov.pop(key)
             for subaction in prov['subactions']:
                 subaction.pop("commit")
+                if subaction["name"] != "GenomeFileUtil":
+                    version = subaction.pop("ver")
+                    assert _PROV_SUBACTION_VERSION_PATTERN.match(version)
         return provs
 
-    def _retrieve_genome_metadata(self, metadata):
-        # make a copy to avoid modifying the original metadata
-        metadata = dict(metadata)
-        metadata.pop("Assembly Object")
+    def _check_assembly_upa(self, retrieved_upa, expected_upa):
+        assert _UPA_PATTERN.match(retrieved_upa)
+        assert retrieved_upa == expected_upa
+
+    def _retrieve_genome_metadata(self, metadata, expected_assembly_upa):
+        # make a deep copy to avoid modifying the original metadata
+        metadata = deepcopy(metadata)
+        retrieved_assembly_upa = metadata.pop("Assembly Object")
+        self._check_assembly_upa(retrieved_assembly_upa, expected_assembly_upa)
         return metadata
 
     def _retrieve_genome_data(self, data):
-        # make a copy to avoid modifying the original genome data
-        data = dict(data)
-        for key in ["assembly_ref", "genbank_handle_ref"]:
-            data.pop(key)
+        # make a deep copy to avoid modifying the original genome data
+        data = deepcopy(data)
         for key in ["cdss", "features", "mrnas", "non_coding_features"]:
             for dist in data.get(key):
-                dist.pop("aliases", None)
+                if dist.get("aliases"):
+                    dist["aliases"] = sorted(dist["aliases"])
+
+        assembly_ref = data.pop("assembly_ref")
+        assert _UPA_PATTERN.match(assembly_ref)
```

Comment:
Need to check this upa is the same as the assembly object upa

Reply:
👍

Reply:
:+1: